crates/primitives/src/transaction/signature.rs

```diff
@@ -62,6 +62,11 @@ impl Signature {
     /// Output the `v` of the signature depends on chain_id
     #[inline]
     pub fn v(&self, chain_id: Option<u64>) -> u64 {
+        #[cfg(feature = "optimism")]
+        if self.r == U256::ZERO && self.s == U256::ZERO {
```

Comment:
this should work now
```suggestion
        if self.r.is_zero() && self.s.is_zero() {
```